c++/include/orc/MemoryPool.hh

```diff
@@ -72,6 +72,10 @@ namespace orc {
       return currentCapacity;
     }
 
+    const T& operator[](uint64_t i) const {
+      return buf[i];
+    }
+
```

Comment:
Does this part have a big impact?

Reply:
The DataBuffer class already has a non-const operator() function, so adding a const operator() function should have no effect.

Reply:
suggest to use the trick to remove duplicates between const and non-const overloads: https://stackoverflow.com/questions/123758/how-do-i-remove-code-duplication-between-similar-const-and-non-const-member-func

Reply:
I write a simple demo for testing the implement overload that use const function to  implement non-const function. 
The complex implement(return const_cast<T&>(static_cast<const DataBuffer<T>&>(*this)[i])) will cost twice the overhead of the simple implement(return buf[i]).
![simple_demo](https://user-images.githubusercontent.com/13637742/168953847-3beb0994-702b-4239-81da-9512c7d56dff.png)

Reply:
Thanks. Could you share your environment information and the numbers with the above, @coderex2522 ?

Reply:
Test Environment: x86_64, gcc version 4.8.5
The test results were 7.03 seconds and 3.82 seconds respectively. @dongjoon-hyun

Reply:
Which OS are you on? The build chain looks a little old to me. I expected `9.4.0` or `11.2.0`.

Reply:
I retested it with gcc 9.2.1 and the results were similar. And both tests were on Linux OS x86_64.